config.go

```diff
@@ -117,9 +119,60 @@ func makeDockerV2S2Image(oimage *ociv1.Image) (docker.Image, error) {
 	return image, nil
 }
 
+func makeDockerV2S1Image(manifest docker.V2S1Manifest) (docker.V2Image, error) {
+	// Treat the most recent (first) item in the history as a description of the image.
+	if len(manifest.History) == 0 {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from manifest")
+	}
+	dimage := docker.V2Image{}
+	err := json.Unmarshal([]byte(manifest.History[0].V1Compatibility), &dimage)
+	if err != nil {
+		return docker.V2Image{}, err
+	}
+	if dimage.DockerVersion == "" {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from history")
+	}
+	// Build a filesystem history.
+	rootFS := &docker.V2S2RootFS{
+		Type:    docker.TypeLayers,
+		DiffIDs: []digest.Digest{},
+	}
+	history := []docker.V2S2History{}
+	for i := range manifest.FSLayers {
+		d := manifest.FSLayers[i].BlobSum
+		// Prepend this layer, because the original manifest's list has the earliest layer last.
+		rootFS.DiffIDs = append([]digest.Digest{d}, rootFS.DiffIDs...)
```

Comment:
Ah, so that explains the discrepancy in https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-1.md.  Fixing to decode v1 compat data correctly.